inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp

```diff
@@ -409,6 +442,21 @@ struct jit_uni_interpolate_kernel_f32 : public jit_uni_interpolate_kernel, publi
     }
 
     void linear_onnx_c_gathered() {
+        mov(reg_dst, ptr[reg_params + GET_OFF(dst)]);
+        mov(reg_src, ptr[reg_params + GET_OFF(weight)]);
+        mov(reg_src_aux, ptr[reg_params + GET_OFF(weightR)]);
+        mov(reg_src_aux1, ptr[reg_params + GET_OFF(weightT)]);
+        mov(reg_src_aux2, ptr[reg_params + GET_OFF(weightB)]);
```

Comment:
In fact approach with separate names for each pixel looks questionable for me. I agree that it makes code easier to read, but in fact it non-scalable. Just image how painfull will be 3rd spatial dimension support. I would recommend to have static arrays of size 4 for src and weights pointers. This will allow to write all repeated code lines as single loop. See https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.h#L82 as as example.

Reply:
modified. Thank you!